pkg/api/models.go

```diff
@@ -52,6 +52,7 @@ const (
 	MessageTypeUserInputResponse  MessageType = "user-input-response"
 	MessageTypeUserChoiceRequest  MessageType = "user-choice-request"
 	MessageTypeUserChoiceResponse MessageType = "user-choice-response"
+	MessageTypeUsageMetadata      MessageType = "usage-metadata"
```

Comment:
We can avoid adding message type to communicate this. I am proposing we should add an attribute on the session.